src/ext/LineSearchesExt.jl

```diff
@@ -0,0 +1,278 @@
+
+"""
+    LineSearchesStepsize <: Stepsize
+
+Wrapper for line searches available in the `LineSearches.jl` library.
+
+## Constructors
+
+    LineSearchesStepsize(
+        M::AbstractManifold,
+        linesearch,
+        initial_stepsize;
+        retraction_method::AbstractRetractionMethod=default_retraction_method(M),
+        vector_transport_method::AbstractVectorTransportMethod=default_vector_transport_method(M),
+    )
+    LineSearchesStepsize(
+        linesearch,
+        initial_stepsize;
+        retraction_method::AbstractRetractionMethod=ExponentialRetraction(),
+        vector_transport_method::AbstractVectorTransportMethod=ParallelTransport(),
+    )
+
+Wrap `linesearch` (for example [`HagerZhang`](https://julianlsolvers.github.io/LineSearches.jl/latest/reference/linesearch.html#LineSearches.HagerZhang)
+or [`MoreThuente`)(https://julianlsolvers.github.io/LineSearches.jl/latest/reference/linesearch.html#LineSearches.MoreThuente))
+and initial step selector `initial_stepsize` that will work on manifold `M`. Retraction used
+for determining the line along which the search initial_stepsize performed can be provided
+as `retraction_method`. Gradient vectors are transported between points using
+`vector_transport_method`.
+"""
+struct LineSearchesStepsize{
```

Comment:
maybe as my interpretation for the storage. This struct would have a storage field (not be wrapped in a storage), the constructor with M can use the fast storage the other one can not. That is the analogue of what other storage things currently do.

Maybe yours is a bit faster that you have here, but I do not see all implications (and do not want to rewrite all high-level interfaces at the moment) and would like to avoid having multiple solutions for one thing. Similarly I do not like that the solver had to be rewritten at all – it uses a stepsize, it does not have to know the internals of said stepsize

Reply:
This is a good tradeoff I think. We could aa a “not supported _yet_” here, since with a proper storage we would do that.

Reply:
Sure, feel free to update the docs if you think so.

Reply:
Sure will check that later today hopefully.